sdk/core/azure-core-experimental/azure/core/experimental/transport/_pyodide.py

```diff
@@ -0,0 +1,158 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+from collections.abc import AsyncIterator
+from io import BytesIO
+
+import js # pylint: disable=import-error
+from pyodide import JsException  # pylint: disable=import-error
+from pyodide.http import pyfetch  # pylint: disable=import-error
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline import Pipeline
+from azure.core.utils import CaseInsensitiveDict
+
+from azure.core.rest._http_response_impl_async import AsyncHttpResponseImpl
+from azure.core.pipeline.transport import HttpRequest, AsyncioRequestsTransport
+
+
+class PyodideTransportResponse(AsyncHttpResponseImpl):
+    """Async response object for the `PyodideTransport`."""
+
+    @property
+    def js_stream(self):
```

Comment:
does this need to be a publicly exposed property? Ideally we want the shape to be the same as `azure.core.rest.AsyncHttpReponse` so we don't have to expose this response